src/shell/server.cpp

```diff
@@ -486,15 +487,23 @@ server::cmd_res server::handle_sync(server::cmd_req const & req) {
     } else {
         new_content = load_module(new_file_name, /* can_use_olean */ false)->m_contents;
     }
-
-    auto mtime = time(nullptr);
+    unsigned new_hash;
+    {
+        std::string tmp = new_content;
+        remove_cr(tmp);
+        // TODO(Vtec234): by how much is hashing on sync going to slow the server down?
+        // The hash should be super-fast -- maybe xxHash?
+        new_hash = hash_data(tmp);
+    }
 
     bool needs_invalidation = true;
 
     auto & ef = m_open_files[new_file_name];
+    // Why not compare hashes here? Because if an editor changes line endings, the hash won't change
```

Comment:
To answer your question: we ignore the line-endings in the `module_mgr` because the line-endings from the on-disk file in some cases differ from the line endings that vscode sends.  Hence we normalize them so that people on windows don't rebuild mathlib after go-to-definition.

In this function here, the `m_open_files` map only contains files sent by vscode.  Hence they should all have the same line endings and we don't need to normalize anything.

But it's completely fine to normalize or compare hashes here.  Byte offsets are completely irrelevant.  And the line ending normalization doesn't change `(line_number, column)`.

Reply:
So we can speed up by a constant factor :) I think I still need to normalize line endings, since this `ef.m_src_hash` may be compared against one computed from a file stored on-disk.